plugins/com.google.cloud.tools.eclipse.appengine.login.test/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginServiceUiTest.java

```diff
@@ -103,8 +103,13 @@ public void run() {
         ) {
           String input = new String();
           while (true) {
-            input += (char) reader.read();
-            if (input.endsWith("?error=cancelled-by-user")) {
+            int ch = reader.read();
+            if (ch == -1) {
+              break;
+            }
+
+            input += (char) ch;
+            if (input.endsWith("?error=canceled-by-user")) {
               cancelRequestReceived = true;
               writer.write("HTTP/1.1 200 OK\n\n");
```

Comment:
That's not a problem; the both sides of the connection can read and write simultaneously through separate input and output streams as long as they are open by the both sides.